pkg/db/canary.go

```diff
@@ -43,7 +43,11 @@ func GetAllCanaries() ([]v1.Canary, error) {
 		return nil, err
 	}
 	for _, _canary := range _canaries {
-		canaries = append(canaries, *_canary.ToV1())
+		c := _canary.ToV1()
+		if c == nil {
+			continue
```

Comment:
@moshloop Should I just return an error here and track it to job history too, the only way this arises is when spec read fails ?

Reply:
We might need to maintain a error list and then return that together with the successful parses